packages/vite-plugin-svelte/src/utils/vite-plugin-svelte-stats.ts

```diff
@@ -0,0 +1,208 @@
+	logInProgress: (c, now) => now - c.collectionStart > 500 && c.stats.length > 1,
+	// always log results
+	logResult: () => true
+};
+
+function humanDuration(n: number) {
+	// 99.9ms  0.10s
+	return n < 100 ? `${n.toFixed(1)}ms` : `${(n / 1000).toFixed(2)}s`;
+}
+
+function formatPackageStats(pkgStats: PackageStats[]) {
+	const statLines = pkgStats.map((pkgStat) => {
+		const duration = pkgStat.duration;
+		const avg = duration / pkgStat.files;
+		return [pkgStat.pkg, `${pkgStat.files}`, humanDuration(duration), humanDuration(avg)];
+	});
+	statLines.unshift(['package', 'files', 'time', 'avg']);
+	const columnWidths = statLines.reduce(
+		(widths: number[], row) => {
+			for (let i = 0; i < row.length; i++) {
+				const cell = row[i];
+				if (widths[i] < cell.length) {
+					widths[i] = cell.length;
+				}
+			}
+			return widths;
+		},
+		statLines[0].map(() => 0)
+	);
+
+	const table = statLines
+		.map((row: string[]) =>
+			row
+				.map((cell: string, i: number) => {
+					if (i === 0) {
+						return cell.padEnd(columnWidths[i], ' ');
+					} else {
+						return cell.padStart(columnWidths[i], ' ');
+					}
+				})
+				.join('\t')
+		)
+		.join('\n');
+	return table;
+}
+
+export class VitePluginSvelteStats {
+	// package directory -> package name
+	private _packages: { path: string; name: string }[] = [];
+	private _collections: StatCollection[] = [];
+	startCollection(name: string, opts?: Partial<CollectionOptions>) {
+		const options = {
+			...defaultCollectionOptions,
+			...opts
+		};
+		const stats: Stat[] = [];
+		const collectionStart = performance.now();
+		const _this = this;
+		let hasLoggedProgress = false;
+		const collection: StatCollection = {
+			name,
+			options,
+			stats,
+			collectionStart,
+			finished: false,
+			start(file) {
+				if (collection.finished) {
+					throw new Error('called after finish() has been used');
+				}
+				const start = performance.now();
+				const stat: Stat = { file, start, end: start };
+				return () => {
+					const now = performance.now();
+					stat.end = now;
+					stats.push(stat);
+					if (!hasLoggedProgress && options.logInProgress(collection, now)) {
+						hasLoggedProgress = true;
+						log.info(`${name} in progress ...`);
+					}
+				};
+			},
+			async finish() {
+				await _this._finish(collection);
+			}
+		};
+		_this._collections.push(collection);
+		return collection;
+	}
+
+	public async finishAll() {
+		await Promise.all(this._collections.map((c) => c.finish()));
+	}
+
+	private async _finish(collection: StatCollection) {
+		collection.finished = true;
+		const now = performance.now();
+		collection.duration = now - collection.collectionStart;
+		const logResult = collection.options.logResult(collection);
+		if (logResult) {
+			await this._aggregateStatsResult(collection);
+			log.info(`${collection.name} done.`, formatPackageStats(collection.packageStats!));
+		}
+		//cut some ties to free it for garbage collection
```

Comment:
the other comments have a space after `//`
```suggestion
		// cut some ties to free it for garbage collection
```